### ISSUE TITLE
Started migration to Scala 3 and ZIO

### ISSUE THREAD

**Issue by carlos-verdes** [avokka/avokka#18]
I have started to port your code in a different repo using ZIO and Scala 3
Different approach with your base code, main changes:
- all code is based on Scala 3 (not planning to support Scala 2)
- I moved from tagless final to ZIO (this means dependencies to that library, at some point I may create the protocol+model only with tagless final and then specific ZIO modules)
- All the protocol now it's abstracted on Encoder[_], Decoder[_] to allow support for both VPack and JSON
- Current implementation with HTTP + JSON (some VPackEncoder started)
- using ZIO-streams instead of fs2 (in general ZIO libraries instead of cats)

Let me know if you are interested to join on this :)

**Comment by gbicou** [comment]
yes sure ! i'm also migrating some projects to scala 3 and using ZIO much more.

**Comment by oridag** [comment]
This is something I definitely want to keep my eye on, I might even find some time to contribute. Please share a repo once there's one up!

**Comment by carlos-verdes** [comment]
I started some code on this repo:
https://github.com/carlos-verdes/funkode

It's a full working code with Scala3, ZIO + HTTP + JSON but I plan to create a new clean repo only for this (I also recreated the testcontainers using ZIO ZLayers)

You can check what is working on this IT test:
https://github.com/carlos-verdes/funkode/blob/scala3/testcontainers-zio2-arangodb/src/it/scala/io/funkode/docker/arangodb/ArangoDbClientIT.scala


One of the big issues is I started using zio-http which is still not released on Maven
There are a lot of tests pending, more VPack support (I implemented the minimum needed to bind vars and do upserts) and some performance test (from ArangoDB team the performance of http+json versus socket+velocypack should not be a big difference but the idea is to implement both options and compare properly with a performance test)

**Comment by carlos-verdes** [comment]
I created new repo where I'll move slowly my code:
https://github.com/carlos-verdes/zio-arangodb

You can participate here (once I start to port the code :))

**Comment by carlos-verdes** [comment]
I have a working code already, and with the release of zio-http 0.0.3 I think I can release the library on Sonatype

**Comment by gbicou** [comment]
nice, I will check this !

**Comment by carlos-verdes** [comment]
Released first version:
```scala
libraryDependencies += "io.funkode" %% "zio-arangodb-http" % "0.0.3"
```
